3-enrich/scala-common-enrich/src/main/scala/com.snowplowanalytics.snowplow.enrich/common/enrichments/registry/AnonIpEnrichment.scala

```diff
@@ -52,23 +56,30 @@ object AnonIpEnrichment extends ParseableEnrichment {
    *        Must be a supported SchemaKey for this enrichment
    * @return a configured AnonIpEnrichment instance
    */
-  def parse(config: JValue, schemaKey: SchemaKey): ValidatedNelMessage[AnonIpEnrichment] =
-    isParseable(config, schemaKey).flatMap(conf => {
+  def parse(config: JValue, schemaKey: SchemaKey): ValidatedNelMessage[AnonIpEnrichment] = {
+
+    def extractParam(name: String) =
+      ScalazJson4sUtils.extract[Int](config, "parameters", name)
+
+    isParseable(config, schemaKey).flatMap(_ => {
       (for {
-        param  <- ScalazJson4sUtils.extract[Int](config, "parameters", "anonOctets")
-        octets <- AnonOctets.fromInt(param)
-        enrich = AnonIpEnrichment(octets)
+        paramIPv4Octet   <- extractParam("anonOctets")
+        paramIPv6Segment <- extractParam("anonSegments").orElse(paramIPv4Octet.success)
+        ipv4Octets       <- AnonIPv4Octets.fromInt(paramIPv4Octet)
+        ipv6Segment      <- AnonIPv6Segments.fromInt(paramIPv6Segment)
+        enrich = AnonIpEnrichment(ipv4Octets, ipv6Segment)
       } yield enrich).toValidationNel
     })
+  }
 
 }
 
 /**
- * How many octets to anonymize?
+ * How many octets (ipv4) to anonymize?
  */
-object AnonOctets extends Enumeration {
+object AnonIPv4Octets extends Enumeration {
 
-  type AnonOctets = Value
+  type AnonIPv4Octets = Value
```

Comment:
In the configuration you have `anonOctets` and `anonSegments` and in the code `AnonIPv4Octets` and `AnonIPv6Segments`, could be more consistent.

Reply:
I have deliberately left config name as it was before (`anonOctets`, and added in a consistent way `anonSegemnts`), while in code I wanted to describe it in more precise way, so I added `IPv4` and `IPv6` to the names.